### PR TITLE
WT-13136 Correct some issues in arch-concurrency.dox and portability.dox

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by luke-pearson** [comment]
I don't know why but I can't request your review directly @RedBeard0531

**Comment by luke-pearson** [comment]
Unresolved comment thread: https://github.com/wiredtiger/wiredtiger/commit/7429bc4a9aa94baa7ff707ae80765d08026fcd77#r142998614

**Review by RedBeard0531 — COMMENTED** [review]
Changes to arch-concurrency.dox look fine. However I find it hard to sign off on a PR to "Correct some issues in arch-concurrency.dox and portability.dox" when it doesn't include the changes to portability.dox.

File: src/docs/arch-concurrency.dox [hunk 1]
```diff
@@ -22,14 +22,16 @@ There are two types of locks in WiredTiger:
  - Read Write (RW) Locks
 
 @subsection lock-usage Lock usage
-A correctly used lock takes the lock for each access to the shared memory it guards. WiredTiger
-sometimes deviates from this model and reads variables outside the context of the relevant lock.
-Such instances are only permitted if reading inconsistent data is sufficient for the use case.
+A correctly used lock must be held when accessing any of the shared memory locations that it
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply src/docs/arch-concurrency.dox:25]: I think access intuitively means read or write so I'll leave it as is.

**Review by RedBeard0531 — APPROVED** [review]

**Review by ajmorton — APPROVED** [review]

**Pull request merged**